SwiftGit2/CommitIterator.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Created by Arnon Keereena on 4/28/17.
+// Copyright (c) 2017 GitHub, Inc. All rights reserved.
+//
+
+import Result
+import libgit2
+
+public class CommitIterator: IteratorProtocol {
+    public typealias Element = Result<Commit, NSError>
+    var repo: Repository
+    var branch: Branch
+    var revisionWalker: OpaquePointer? = nil
+    var oid: git_oid
+    private var unsafeCommit: OpaquePointer? = nil
+    
+    public init(repo: Repository, branch: Branch) {
```

Comment:
Let's make this `internal`. People should create iterator using `Repository.commits(in: Branch)`. We'll also want to create different types of iterators, so this is unnecessarily restrictive.